kafka-ui-api/src/main/java/com/provectus/kafka/ui/config/OAuthSecurityConfig.java

```diff
@@ -41,9 +43,28 @@
   };
 
   private final ApplicationContext context;
+  private final Boolean cognitoEnabled;
+  private final String clientId;
+  private final String logoutUrl;
+
+  public OAuthSecurityConfig(final ApplicationContext applicationContext,
+                             @Value("${auth.cognito.enabled}") final Boolean cognitoEnabled,
+                             @Value("${spring.security.oauth2.client.registration.cognito.client-id:}")
+                             final String clientId,
+                             @Value("${auth.cognito.logoutUrl:}") final String logoutUrl) {
+    this.context = applicationContext;
+    this.cognitoEnabled = cognitoEnabled;
+    this.clientId = clientId;
+    this.logoutUrl = logoutUrl;
+  }
 
   @Bean
-  public SecurityWebFilterChain configure(ServerHttpSecurity http) {
+  public SecurityWebFilterChain configure(ServerHttpSecurity http, final ReactiveClientRegistrationRepository repo) {
+
+    final ServerLogoutSuccessHandler logoutHandler = cognitoEnabled
+        ? new CognitoOidcLogoutSuccessHandler(logoutUrl, clientId)
```

Comment:
I don't like the idea to have such configs.

Reply:
The same as in ODD as well, though. What do you suggest? We don't have any other markers to identify if the user's using cognito or not besides this introduced variable.

Reply:
done